# 1. Реализуйте метод, который запрашивает у пользователя ввод дробного числа (типа float), и возвращает введенное значение.
# Ввод текста вместо числа не должно приводить к падению приложения,
# вместо этого, необходимо повторно запросить у пользователя ввод данных.

def main():
    try:
        tepm = float(input("Введите добное число: "))
        return tepm
    except ValueError:
        print("Вы ввели не допустимые символы")
        return main()

# 2. Если необходимо, исправьте данный код

def les2():
    intArray = [0,0,0,0,0,0,0,0,0]
    try:
        d: int = 1
        catchedRes1: float = intArray[8] / d
        print(f"{catchedRes1 = }")
    except ZeroDivisionError as e:
        print("Catching exception:", e)

# 3.Дан следующий код, исправьте его там, где требуется
# (задание 3 https://docs.google.com/document/d/17EaA1lDxzD5YigQ5OAal60fOFKVoCbEJqooB9XfhT7w/edit)
def les2_3():
    abc = [0,0,0,0]
    try:
        a: int = 90
        b: int = 3
        print(a/b)
        printSum(23, 234)
        abc[3] = 9
    except TypeError:
        print("Указатель не может указывать на null!")
    except IndexError:
        print("Массив выходит за пределы своего размера!")
    except ZeroDivisionError:
        print("Деление на ноль")
    except Exception:
        print("Что-то пошло не так...")


def printSum(num1, num2):
    return print(num1+num2)

#4. Разработайте программу, которая выбросит Exception, когда пользователь вводит пустую строку.
# Пользователю должно показаться сообщение, что пустые строки вводить нельзя.
def les2_4():
    try:
        temp = input("Введите любой текст: ")
        if temp:
            print(temp)
        else:
            raise error_my("Пустая строка, введите текст")
    except error_my as e:
        print("Ошибка:", e.message)
class error_my(Exception):
    def __init__(self, message):
        self.message = message
        super().__init__(self.message)


if __name__ == '__main__':
    main()
    les2()
    les2_3()
    les2_4()
